docs/edgeql/operators/json.rst

```diff
@@ -39,17 +39,51 @@ JSON values can also be cast back into scalars. This casting is
 symmetrical meaning that if a scalar can be cast into JSON, only that
 particular JSON type can be cast back into that scalar:
 
-- JSON *string* can be cast into :eql:type:`str`
+- JSON *string* can be cast into :eql:type:`str`. Casting
+  :eql:type:`uuid` and :ref:`date and time types
+  <ref_datamodel_scalars_datetime>` to JSON results in a JSON
+  *string* representing the original value. This means that it is
+  also possible to cast a JSON *string* back into these types. The
+  string value has to be properly formatted (much like in case of
+  a :eql:type:`str` value being cast) or else the cast will raise an
+  exception.
 - JSON *number* can be cast into any of
   the :ref:`numeric types <ref_datamodel_scalars_numeric>`
 - JSON *boolean* can be cast into :eql:type:`bool`
 - JSON *null* is special since it can be cast into an ``{}`` of any type
 - JSON *array* can be cast into any valid EdgeDB array, so it must be
   homogeneous, and must not contain *null*
-- JSON *object* can be cast into a valid EdgeDB named tuple, but it must not
-  contain *null*
 
-The contents of JSON *objects* and *arrays* can also be accessed via ``[]``:
+*Unnamed* :eql:type:`tuple` is converted into a JSON *array* when cast
+into :eql:type:`json`. Whereas *named* :eql:type:`tuple` is converted
+into a JSON *object*. These casts are not reversible, i.e. it is not
+possible to cast a JSON value directly into a :eql:type:`tuple`.
+
+
+Casting Objects into JSON
+=========================
+
+Any :eql:type:`Object` can be cast into :eql:type:`json`. This
+produces the same JSON value as the JSON serialization of that object.
+That is, the result is the same as the output of :ref:`SELECT
+expression<ref_eql_statements_select>` in *JSON mode*, including the
+type shape.
+
+.. code-block:: edgeql-repl
+
+    db> WITH MODULE schema
+    ... SELECT <json>(Type {
+    ...     name,
+    ...     timestamp := <naive_date>datetime_current()
+    ... })
+    ... FILTER Type.name = 'std::bool';
+    {{name: 'std::bool', timestamp: '2019-01-18'}}
+
+
+Accessing JSON Array Elements
+=============================
```

Comment:
Maybe "Accessing Elements of JSON Arrays" sounds better? @elprans @vpetrovykh

Reply:
Not a fan to he honest. "JSON array element" and "JSON object field" is actually a bit easier to read than "element of JSON array" or "field of JSON object". It's the same reason that "accessing foo.bar.baz" reads easier than "accessing baz of bar of foo".